lib/broccoli/ember-app.js

```diff
@@ -388,12 +385,8 @@ class EmberApp {
     @return
   */
   concatFiles(tree, options) {
-    this.project.ui.writeDeprecateLine('EmberApp.concatFiles() is deprecated. Please use the `broccoli-concat` module directly.',
-      arguments[2] === SECRET_DEPRECATION_PREVENTION_SYMBOL);
-
-    options.sourceMapConfig = this.options.sourcemaps;
-
-    return concat(tree, options);
+    this.project.ui.writeDeprecateLine('EmberApp.concatFiles() is deprecated. Please use the `broccoli-concat` module directly.');
```

Comment:
This unconditionally deprecates calling `this.concatFiles` (and I presume refactors the internals to avoid calling it and instead call `_concatFiles`). We originally still funneled work through `this.concatFiles` because of a ember-cli-fastboot [monkey patch](https://github.com/ember-fastboot/ember-cli-fastboot/blob/ee4bd3b2e32a8b3995a18322ef9fbd25e05047cd/lib/ext/patch-ember-app.js#L20-L25).  I'm happy to remove this, but we need to also update that monkey patch to continue to function after this lands.

FWIW, the fastboot team has a path forward to remove the need for this monkey patch, and @kratiahuja has been working hard on landing. Take a peek at the [meeting summary](https://gist.github.com/kratiahuja/fd073007e10abb9db0a2ec42bc1d7c17)  where we came to consensus...